core/spec.md

```diff
@@ -847,45 +849,65 @@ In API view:
 
 ##### Updating Nested Registry Collections
 
 When updating an entity that might contain Registry collections, the request
 MAY contain the 3 collection attributes. The `COLLECTIONsurl` and
 `COLLECTIONscount` attributes MUST be silently ignored by the server.
-If the `inline` query parameter is not present, then the `COLLECTIONs`
-attribute MUST also be silently ignored by the server.
+By default, in the absence of a `nested` query parameter, the server MUST
+ignore the `COLLECTIONs` attribute as well.
 
-Note: in the [Inlining](#inlining) section of the specification, `inline`
-is defined such that it might include a `PATH` value. That is only used for
-retrieval (`GET`) operations. On write operations, the `inline` query
-parameter does not define any values and therefore MUST be treated as a
-boolean flag.
-
-If the `inline` query parameter and the `COLLECTIONs` attribute is present,
+If the `nested` query parameter and the `COLLECTIONs` attribute are present,
 the server MUST process each entity in the collection map as a request to
-create or update that entity based on whether it already exists (as determined
-by its key value - its `id`). An entry in the map that isn't a valid entity
-(e.g. is `null`) MUST generate an error.
+create or update that entity according to the semantics of the HTTP method used.
+An entry in the map that isn't a valid entity (e.g. is `null`) MUST generate
+an error.
+
+The `nested` semantics MUST be applied to all nested `COLLECTIONs` attributes
+within the request regardless of the level in the hierarch in which they appear.
+
+For example:
 
-The `inline` semantics MUST be applied to all nested `COLLECTIONs` attributes
-within the request regardless of the level in the hierarch in which they
-appear.
+```
+PUT https://example.com/endpoints/123?nested
 
-When processing a nested entry in the collection, it MUST be processed using
-the same semantics as the top-most entity. In other words, if a `PUT` or
-`POST` was used then any entities in the collection will be fully replaced. If
-a `PATCH` was used then only the attributes that appear in the serialization
-of the nested entity will be processed.
+{
+  "id": "123",
+  "name": "A cool endpoint",
+
+  "definitions": {
+    "mydef1": { ... },
+    "mydef2:" { ... }
+  }
+}
+```
+
+Will not only create/update an `endpoint` Group with an `id` of `123` but will
+also create/update its `definition` Resources (`mydef1` and `mydef2`).
 
 Any error while processing a nested collection entity MUST result in the entire
 request being rejected.
 
 An absent `COLLECTIONs` attribute MUST be interpreted as a request to not
-modify the collection at all due to processing of the request, regardless of
-the presence (or absence) of the `inline` query parameter.
+modify the collection at all, regardless of the presence (or absence) of the
+`nested` query parameter.
+
+If a client wishes to replace an entire collection, rather than just add new
```

Comment:
So this means it can't be done atomically, right?

Reply:
Correct. Do you see an issue with that? We were trying to keep it simple but if you think it causes pain we could revisit this decision.

Reply:
My lack of knowledge of use cases means it's hard to say for sure. But I can imagine it *potentially* being a problem. Probably something worth just checking with other folks.

Reply:
On the positive side, if we wanted to support it I think it would be via a PUT to the collection, which as of now results in an error, so we could use that method as a "complete collection replacement". But we'll discuss!